### PR TITLE
Blazor and Angular data hierarchy.

### PR DESCRIPTION
**Pull request by atmgrifter00**
# Pull Request

## 🤨 Rationale

- #861

## 👩‍💻 Implementation

Just adding standard Angular and Blazor integration APIs (a single Table attribute).

## 🧪 Testing

Added standard unit tests. Updated example apps.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.ts [hunk 1]
```diff
@@ -120,6 +121,7 @@ export class CustomAppComponent {
         for (let i = 0; i < numberOfRowsToAdd; i++) {
             tableData.push({
                 id: tableData.length.toString(),
+                parentId: (tableData.length > 4 ? ((tableData.length + i) % 4).toString() : undefined),
```
Comment (m-akinc) [inline angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.ts:124]: This seems more like the intent:
```suggestion
                parentId: (tableData.length >= 4 ? (tableData.length % 4).toString() : undefined),
```

File: angular-workspace/projects/ni/nimble-angular/table/tests/nimble-table.directive.spec.ts
Comment (m-akinc) [inline angular-workspace/projects/ni/nimble-angular/table/tests/nimble-table.directive.spec.ts:1]: Should add "has expected defaults for parentIdField" test case for completeness. Also, I notice we don't have any "with template string values" tests like we do for other directives. Consider adding them for the few properties that apply.

File: packages/nimble-blazor/Examples/Demo.Shared/Pages/ComponentsDemo.razor.cs [hunk 2]
```diff
@@ -68,6 +68,7 @@ public void AddTableRows(int numberOfRowsToAdd)
 
                 tableData.Add(new SimpleTableRecord(
                     rowCountString,
+                    tableData.Count > 4 ? ((tableData.Count + i) % 4).ToString(CultureInfo.CurrentCulture) : null,
```
Comment (m-akinc) [inline packages/nimble-blazor/Examples/Demo.Shared/Pages/ComponentsDemo.razor.cs:71]: If the other suggestion was accepted, then:
```suggestion
                    tableData.Count >= 4 ? (tableData.Count % 4).ToString(CultureInfo.CurrentCulture) : null,
```

**Review by m-akinc — APPROVED** [review]

**Review by rajsite — APPROVED** [review]

**Comment by rajsite** [comment]
@msmithNI can you take a look as a blazor code owner?

**Review by msmithNI — APPROVED** [review]

**Pull request merged**